cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -322,36 +314,59 @@ def _upload_fit_artifact(
         tune_data,
         config,
         serving_script,
-        images=None,
+        image_column=None,
     ):
         cloud_bucket, cloud_key_prefix = s3_path_to_bucket_prefix(self.cloud_output_path)
         util_key_prefix = cloud_key_prefix + "/utils"
+
+        common_path = None
+        if image_column is not None:
+            # Find common path to zip and replace image column with relative path to be used in remote environment
+            if isinstance(train_data, str):
+                train_data = load_pd.load(train_data)
+            else:
+                train_data = copy.deepcopy(train_data)
+            if tune_data is not None:
+                if isinstance(tune_data, str):
+                    tune_data = load_pd.load(tune_data)
+                else:
+                    tune_data = copy.deepcopy(tune_data)
+            common_train_data_path = os.path.commonpath(train_data[image_column].tolist())
```

Comment:
We should check if common_path is the same root path such as `/`. Otherwise, we probably will zip the whole volume and trigger `out of disk space` issue

Reply:
synced offline. Mark as resolved